microsoft/PowerToys
# Adds GcodeThumbnailProvider and GcodePreviewHandler

## Summary of the Pull Request

**What is this about:**

Adds a new preview handler and thumbnail provider for gcode (3D printer files).

The gcode must already have the thumbnail inline.

Screenshot of sample file:

![image](https://user-images.githubusercontent.com/85504/144943043-5a7a70f1-d19d-40c3-9ed0-fc2f8fa82b40.png)

Screenshot of the settings page:

![image](https://user-images.githubusercontent.com/85504/145020415-b544725a-06d3-4e9d-af36-b37d03d727d5.png)

**What is include in the PR:** 

A new preview handler and thumbnail provider and unit tests for both.

**How does someone test / validate:** 

- Close PowerToys if installed in your machine
- Full build the solution: PowerToys.sln
- Start PowerToys from PowerToys\x64\Debug\PowerToys.exe or PowerToys\x64\Release\PowerToys.exe
- Toggle the "Gcode thumbnail previewer" setting to enable
- Open HelperFiles folder on the tests and check if the icon changes to an image
- Check explorer preview to see if image is also shown there

## Quality Checklist

- [x] **Linked issue:** #14152 

## Contributor License Agreement (CLA)
A CLA must be signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA.
